src/btree/bt_sync_obsolete.c

```diff
@@ -391,6 +407,24 @@ __checkpoint_cleanup_page_skip(
         return (0);
     }
 
+    if (__sync_obsolete_limit_reached(session)) {
+        *skipp = true;
+        return (0);
+    }
+
+    /*
+     * Read the page on disk if all the content of the page is not fully deleted and the most recent
+     * transaction is globally visible.
+     */
+    newest_ts = WT_MAX(addr.ta.newest_start_durable_ts, addr.ta.newest_stop_durable_ts);
+    if (!WT_TIME_AGGREGATE_HAS_STOP(&addr.ta) &&
+      __wt_txn_newest_visible_all(session, addr.ta.newest_txn, newest_ts)) {
```

Comment:
We should also check if the newest_txn and newest_ts are none or not?

Reply:
It is checked as part of the `__wt_txn_newest_visible_all` function.

Reply:
If that is the case, the name of the function is not right. 0 0 should be visible all.

Reply:
@quchenhao, would `__wt_txn_newest_valid_visible_all` make more sense?

Reply:
May be `__wt_txn_has_newest_and_visible_all`?